File: src/p0078/python/solution.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ansList, ans = [], []
        def dfs(i: int):
            if i == len(nums):
                ansList.append(ans.copy())
                return
            ans.append(nums[i])
            dfs(i+1)
            ans.pop()
            dfs(i+1)
        dfs(0)
        
        return ansList

File: src/p0078/golang/solution.go
func subsets(nums []int) [][]int {
	var (
		dfs func(i int)
		ans [][]int
		t   []int
	)
	dfs = func(i int) {
		ans = append(ans, append([]int(nil), t...))
		for j := i; j < len(nums); j++ {
			t = append(t, nums[j])
			dfs(j + 1)
			t = t[:len(t)-1]
		}
	}
	dfs(0)

	return ans
}